labs/lab5/lab5.md

```diff
@@ -477,10 +502,80 @@ combinazione con una funzione `print_state` per la stampa di:
 ```c++
 std::cout << "Report for each fo the stored states:\n";
 std::cout << " length : center of mass x : center of mass v\n";
-std::for_each(v_states.begin(), v_states.end(), print_state);```
+std::for_each(v_states.begin(), v_states.end(), print_state);
 ```
 
 > __NOTA__: anche all'interno di `print_state` potrebbe essere utile avvalersi
 > di `std::accumulate`. Come potreste fare per gestire l'accumulazione di
 > `sum_m`, `sum_mx`, e `sum_mv` evitando di utilizzare `std::accumulate` più
 > volte?
+
+È vero, questa separazione implica effettuare due _loop_ su `v_states`, uno in
+`print_state` ed uno in `print_summary`, ma per questo laboratorio è un
+compromesso accettabile che offre una scusa per _fare pratica_.
+
+#### Inserimento dei punti della `Chain` tramite `generate_n`
+
+Potete inoltre avvalervi di
+[`std::generate_n`](https://en.cppreference.com/w/cpp/algorithm/generate_n) per
+aggiungere i punti alla `Chain`, sostituendo:
+
+```c++
+  double const p_m{0.05};
+
+  double p_x{0.0};
+  double delta_x{0.011};
+
+  for (auto const p_v : {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0}) {
+    chain.push_back({p_m, p_x, p_v});
+    p_x += delta_x;
+  };
+```
+
+partendo da:
+
+```c++
+  int const n_points{10};
+  double const p_m{0.05};
+  double const p_v{0.0};
+
+  double p_x{0.0};
+  double delta_x{0.011};
+
+  std::generate_n(std::back_inserter(c_hooke), ... );
+```
+
+> __NOTA__: in questo caso, suggeriamo di assumere che `p_v` sia sempre `0.0`.
+>
+> __NOTA__: come primo argomento  di `std::generate_n` potete utilizzare
+> `std::back_inserter(c_hooke)`, che invoca la funzione `Chain::push_back` ad
+> ogni passo dell'iterazione.
+>
+> __NOTA__: Se non l'avete fatto, sarà necessario aggiungere una piccola
+> dichiarazione `using` per evitare errori di compilazione quando si utilizza
```

Comment:
non ho capito questa cosa di usare `using`

Reply:
Ho aggiustato il suggerimento (e risposto in chat)